app/lib/core/models/animal_transport_record.dart

```diff
@@ -17,15 +17,43 @@ class AnimalTransportRecord {
   final DeliveryInfo deliveryInfo;
   final AcknowledgementInfo ackInfo;
   final ContingencyPlanInfo contingencyInfo;
+  final String atrId;
```

Comment:
- [ ] Shall we add the `isComplete` field as well?
The ATR ID is the uniquely identifying ID from firebase right, when we make an ATR, we should set the ID. How can we do this if the ATR is made before it's inserted into the database?

Reply:
Maybe also add `ATRwithID` `ATRwithUserId` methods.

Reply:
- I will add isComplete field in `AnimalTransportRecord` in my next commit
- The atrId's purpose is to check who created that specific record. It uniquely identifies who created the record. 

> when we make an ATR, we should set the ID 
This task will be covered in atr newRecord query. The atrId will be set before inserting the new record.

- Where are we going to use the suggested methods?

Reply:
Ah, right. I got ahead of myself. We won't need to overwrite the ATR or User identifier because these don't change.

- [ ] We should add a `withMarkedComplete()` method to set the isComplete field.

_Recall this ATR is immutable, so we should not make `isComplete` not final to avoid needing `withMarkedComplete()`. We should use `with()`._